icu4c/source/test/cintltst/udatatst.c

```diff
@@ -1157,7 +1157,7 @@ static void TestICUDataName()
     }
 
     /* Only major number is needed. */
-    sprintf(expectDataName, "%s%d%c",
+    snprintf(expectDataName, 20, "%s%d%c",
```

Comment:
Is there a reason why you didn't use `sizeof` here?

Reply:
somehow I missed this